src/main/kotlin/no/fdk/resourceservice/service/RdfService.kt

```diff
@@ -0,0 +1,439 @@
+ */
+@Service
+class RdfService(
+    private val objectMapper: ObjectMapper,
+) {
+    private val logger = LoggerFactory.getLogger(RdfService::class.java)
+
+    /**
+     * Supported RDF formats and their corresponding media types.
+     */
+    enum class RdfFormat(
+        val mediaType: MediaType,
+        val fileExtension: String,
+    ) {
+        JSON_LD(MediaType("application", "ld+json"), "jsonld"),
+        TURTLE(MediaType("text", "turtle"), "ttl"),
+        RDF_XML(MediaType("application", "rdf+xml"), "rdf"),
+        N_TRIPLES(MediaType("application", "n-triples"), "nt"),
+        N_QUADS(MediaType("application", "n-quads"), "nq"),
+    }
+
+    /**
+     * RDF format style options for all RDF serializations.
+     */
+    enum class RdfFormatStyle {
+        /**
+         * Pretty format with namespace prefixes, human-readable.
+         */
+        PRETTY,
+
+        /**
+         * Standard format with namespace prefixes and compact representation.
+         */
+        STANDARD,
+    }
+
+    /**
+     * Determines the best RDF format based on the Accept header.
+     *
+     * @param acceptHeader The value of the Accept HTTP header.
+     * @return The best matching RdfFormat, or JSON_LD if no match or header is null/empty.
+     */
+    fun getBestFormat(acceptHeader: String?): RdfFormat {
+        if (acceptHeader.isNullOrBlank()) {
+            return RdfFormat.JSON_LD
+        }
+
+        val acceptTypes =
+            acceptHeader
+                .split(',')
+                .map { it.trim().lowercase() }
+                .filter { it.isNotBlank() }
+
+        // Prioritize exact matches
+        for (acceptType in acceptTypes) {
+            for (format in RdfFormat.entries) {
+                if (format.mediaType.toString() == acceptType) {
+                    return format
+                }
+            }
+        }
+
+        // Fallback to JSON_LD if no specific match
+        return RdfFormat.JSON_LD
+    }
+
+    /**
+     * Returns the MediaType for a given RdfFormat.
+     *
+     * @param format The RdfFormat.
+     * @return The corresponding MediaType.
+     */
+    fun getContentType(format: RdfFormat): MediaType = format.mediaType
+
+    /**
+     * Converts from Turtle format to any target format.
+     */
+    fun convertFromTurtle(
+        turtleData: String,
+        toFormat: RdfFormat,
+        style: RdfFormatStyle,
+        expandUris: Boolean = false,
+    ): String? {
+        val model = ModelFactory.createDefaultModel()
+        val inputStream = ByteArrayInputStream(turtleData.toByteArray())
+        RDFDataMgr.read(model, inputStream, Lang.TURTLE)
+
+        if (expandUris) {
+            model.clearNsPrefixMap()
+        }
+
+        val rdfFormat = getRdfFormat(toFormat, style)
+        val outputStream = StringWriter()
+        RDFDataMgr.write(outputStream, model, rdfFormat)
+
+        return handleSpecialCases(outputStream.toString(), rdfFormat)
```

Comment:
Fungerer dette? Om du bruker `use` så vil den automatisk kjøre `close()` etterpå
```suggestion
        ModelFactory.createDefaultModel().use { model ->
          val inputStream = ByteArrayInputStream(turtleData.toByteArray())
          RDFDataMgr.read(model, inputStream, Lang.TURTLE)

          if (expandUris) {
              model.clearNsPrefixMap()
          }

          val rdfFormat = getRdfFormat(toFormat, style)
          val outputStream = StringWriter()
          RDFDataMgr.write(outputStream, model, rdfFormat)

          return handleSpecialCases(outputStream.toString(), rdfFormat)
        }
```